pydantic/_internal/_fields.py

```diff
@@ -109,7 +110,23 @@ def collect_model_fields(  # noqa: C901
             continue
         for protected_namespace in config_wrapper.protected_namespaces:
             if ann_name.startswith(protected_namespace):
-                raise NameError(f'Field "{ann_name}" has conflict with protected namespace "{protected_namespace}"')
+                valid_namespaces = tuple(x for x in config_wrapper.protected_namespaces if not ann_name.startswith(x))
```

Comment:
as `valid_namespaces` is used in `else` block, it might be better to move it there to prevent extra loop